server/src/app-dto/enter-attempt.dto.ts

```diff
@@ -1,4 +1,5 @@
-import { IsInt, IsNotEmpty, IsNumber, IsOptional, IsString, Min } from 'class-validator';
+import { IsInt, IsNotEmpty, IsNumber, IsOptional, IsString, Matches, Min } from 'class-validator';
+import C from '~~/client/shared_helpers/constants';
```

Comment:
This is not important, but you can use @sh as an alias for the shared_helpers directory

Reply:
Changed